packages/wrapped-keys/src/lib/wrapped-keys.ts

```diff
@@ -0,0 +1,216 @@
+import { CHAIN_ETHEREUM, ENCRYPTED_PRIVATE_KEY_ENDPOINT } from './constants';
+import { decryptToString, encryptString } from '@lit-protocol/encryption';
+import { logError } from '@lit-protocol/misc';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from './utils';
+import {
+  LitMessage,
+  LitTransaction,
+  ExportPrivateKeyParams,
+  ExportPrivateKeyResponse,
+  ImportPrivateKeyParams,
+  ImportPrivateKeyResponse,
+  SignWithEncryptedKeyParams,
+} from './interfaces';
+
+export async function importPrivateKey({
+  pkpSessionSigs,
+  privateKey,
+  litNodeClient,
+}: ImportPrivateKeyParams): Promise<string> {
+  const firstSessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(firstSessionSig);
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: allowPkpAddressToDecrypt,
+      dataToEncrypt: privateKey,
+    },
+    litNodeClient
+  );
+
+  const data = {
+    ciphertext,
+    dataToEncryptHash,
+  };
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const errorBody = await response.text();
+      logError(
+        `Could not import the encrypted key due to the error: ${errorBody}`
+      );
+
+      throw new Error(errorBody);
+    }
+
+    const importedPrivateKey: ImportPrivateKeyResponse = await response.json();
+    return importedPrivateKey.pkpAddress;
+  } catch (error) {
+    const errorMessage = `There was a problem fetching from the database: ${error}`;
```

Comment:
We might not want to expose the actual database error to the caller since they do not own the resource.

Reply:
It's not the DB error but Lambda error. @MaximusHaximus wdyt

Reply:
Someone can dump the DB, make sure on Lambda that not happen